syntax = "proto2";

message Vision_DetectionBall {
  optional float  vel_x      =  1;
  optional float  vel_y      =  2; 
  optional uint32 area       =  3;
  required float  x          =  4;
  required float  y          =  5;
  optional float  height     =  6;
  optional uint32 ball_state =  7;
  optional uint32 last_touch =  8;
  required bool   valid      =  9;
}

message Vision_DetectionRobot {
  required float  confidence  =  1;
  optional uint32 robot_id    =  2;
  required float  x           =  3;
  required float  y           =  4;
  optional float  orientation =  5;
  optional float  vel_x       =  6;
  optional float  vel_y       =  7; 
  optional float  rotate_vel  =  8;
}

message Vision_DetectionFrame {
  required Vision_DetectionBall  balls         = 1;
  repeated Vision_DetectionRobot robots_yellow = 2;
  repeated Vision_DetectionRobot robots_blue   = 3;
}
